oauth/api/router.py

```diff
@@ -0,0 +1,167 @@
+from logging import getLogger
+from typing import List
+
+from fastapi import APIRouter, Body, Depends, Request, Security, status
+from fastapi.security import HTTPBasic
+from sqlalchemy.orm import Session
+
+import oauth.api.endpoints as endpoints
+from oauth.api.exceptions import (
+    AuthenticationException,
+    ClientNotFoundException,
+    InvalidScopeException,
+)
+from oauth.api.models import (
+    AccessToken,
+    ClientCreatedResponse,
+    OAuth2ClientCredentialsRequestForm,
+)
+from oauth.api.utils import verify_oauth_client
+from oauth.database.models import ClientDetail
+from oauth.scopes import (
+    CLIENT_CREATE,
+    CLIENT_DELETE,
+    CLIENT_READ,
+    CLIENT_UPDATE,
+    SCOPE_READ,
+    SCOPES,
+)
+
+logger = getLogger(__name__)
```

Comment:
I'm on the fence about keeping the logging in this library. Personally, I hate it when imported packages write logs that I can't control. On the other hand, some of these endpoints are performing sensitive actions and should have a non-negotiable audit trail.

Reply:
Pretty sure that the Python logger is always under control of the broader project in some kind of weird and confusing ways, no? Wouldn't we want to make sure that logs are being generated, but that they obey the log levels / etc. set by the outer env?

Reply:
The fact that it's using the default package-level logger is the only reason I kept it. In fidesctl we have a custom logger that wraps any logs written within the API server code, so we have very good control. I can't speak for other projects that might import this library though.

Reply:
Doesn't `getLogger` just mean it'll start using the logger that fidesctl defined, when you import it there?

Reply:
In this instance, `getLogger` is creating a `logging.Logger` specific to this package. In the fidesctl API server, we have code that automatically removes all `logging.Handler`s from all loggers, and routes any log entries created within the API to the root logger. We then handle root logger entries using a custom `logging.Formatter` that we define.

So if we import this package and use it within the fidesctl API server, yes, any log entries created by this logger will be handled by our custom logger. The same would not automatically be true in another codebase; the default handling would take precedence unless log entries are routed/handled elsewhere.